Allow multiple API keys management with profiles

bonidier opened the pull request:
Hi @denouche , Thanks for your script, very useful !

Here, I've resolved some limitations :

-  Your script can use only one pair of Application+Consumer Keys, but we may need to use it for many OVH accounts or uses.
-  When you init keys, it force all grants to the API : It may be dangerous if using your client on a server with automated tasks.

Regards

bonidier committed:
Allow multiple API keys management with profiles
- add --profile to override default configuration location
- add --list-profile to display available profiles
- rework messages to display command with the current profile if set

Bugfix
- ensure OVH App Key an App Secret are defined when creating consumer key

Enhancements
- when using --init, allow override default "all permissions grants" if --data is set
- add -h/--help

bonidier commented:
Hi @denouche , Thanks for your script, very useful !

Here, I've resolved some limitations : 
- Your script can use only one pair of Application+Consumer Keys, but we may need to use it for many OVH accounts or uses.
- When you init keys, it force all grants to the API : It may be dangerous if using your client on a server with automated tasks.


Regards

denouche commented on ovh-api-bash-client.sh:
```diff
@@ -201,28 +229,70 @@ getJSONFieldString()
+    CURRENT_PATH="$( cd "${PROFILES_PATH}/${profile}" && pwd )"
+    HELP_CMD="${HELP_CMD} --profile ${profile}"
+  else
+    echo "profile not set, choose in : "
```
In this case maybe we could use a `default` profile, to avoid these modifications to be breaking for existing users?

bonidier replied:
@denouche this doesn't break users : if --profile is not set, the keys files goes to script location. 
(the CURRENT_PATH in only overrided in the initProfile function)

I can :
- simulate an explicit profile _default_  : if setted to that value, or without --profile xxx
- moving the default location (CURRENT_PATH value) to the initProfile function.

It should be more logical.

denouche commented:
Wow, really nice suggestions!
Thank you for your PR.

I only added 1 comment. Let me know what you think about my suggestion.

bonidier commented on ovh-api-bash-client.sh:
```diff
@@ -18,8 +18,13 @@ API_URLS[EU]="https://api.ovh.com/1.0"
 
+BASE_PATH="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+CURRENT_PATH="${BASE_PATH}"
+#PROFILES_PATH
```
I should remove that

bonidier committed:
if --profile is set to 'default', load configuration from script directory

bonidier commented:
Hi @denouche, I've made changes like expected : the virtual profile "default" is now mapped to script directory (like if you don't set --profile argument)

Regards

bonidier committed:
Allow to set --profile option at any position :
- functions to create keys now launched out of parseArguments()

bugfix :
- fix empty CURRENT_PATH when launching --init/--initApp

enhancements :
- better output for --list-profile output + add 'default' profile

refactoring :
- move initApplication() into createConsumerKey() instead of calling it each time

denouche commented:
Nice.

My last 2 comments:
- if you execute: `$ rm -rf ./profile && ./ovh-api-bash-client.sh --profile test`, the folders listed are the folders in the base path instead of just `default`, the result is:
```
Available profiles : 
- default
- libs
- profile
```

- then if I call `./ovh-api-bash-client.sh --profile test --initApp` it also just list the available profiles and quit. Around line 254, maybe we could create the profile folder if the `INIT_KEY_ACTION` is set? 
That's okay to list the profile and quit in "normal" usage, but when initializing I think it could be nice to create the folder for the user and continue the initialization.

bonidier committed:
profiles management enhancements
- when launching --init/--initApp, create the defined profile if missing
- fix profiles listing (bad directory)
- help command : always add --profile argument if profile name is defined

bonidier commented:
Hi @denouche, thanks for your feedback !

I've solved your requests, It's working fine now.

Good idea for the profile creation when initializing, it avoid the user to run the mkdir before.

Regards

denouche commented:
Thank you for your contribution!

:+1:
